test/Routing/Dht1Test.cs

```diff
@@ -85,10 +86,10 @@ public async Task ProcessFindNodeMessage_Self()
                 var request = new DhtMessage
                 {
                     Type = MessageType.FindNode,
-                    Key = self.Id.ToArray()
+                    Key = ByteString.CopyFrom(self.Id.ToArray()),
                 };
                 var response = dht.ProcessFindNode(request, new DhtMessage());
-                Assert.AreEqual(1, response.CloserPeers.Length);
+                Assert.AreEqual(1, response.CloserPeers.Count);
```

Comment:
Why switch from Length to Count?

Reply:
Length is only available in Array, however, the generated code is type of RepeatedField which implements IList, thus only Count is available

Reply:
Array also implements IList, so it's good practice for a library to expose IList<T> / IReadOnlyList<T> to public as appose to Array itself

Reply:
In general, I agree.  However byte arrays should be treated as an array.